include/wx/textctrl.h

```diff
@@ -126,6 +126,76 @@ enum wxTextCtrlHitTestResult
 };
 // ... the character returned
 
+#if wxUSE_SPELLCHECK
+
+// ----------------------------------------------------------------------------
+// This object can be passed to wxTextCtrl::EnableProofCheck() to configure the
+// proofing options for this control.
+// ----------------------------------------------------------------------------
+class wxTextProofOptions
+{
+public:
+    // Return the object corresponding to the default options: current
+    // language, spell checking enabled, grammar checking disabled.
+    static wxTextProofOptions Default()
+    {
+        wxTextProofOptions opts;
+        return opts.SpellCheck(true);
+    }
+
+    // Return the object with all checks disabled.
+    static wxTextProofOptions Disable()
+    {
+        return wxTextProofOptions();
+    }
+
+    // Default copy ctor, assignment operator and dtor are ok
+
+    // Methods that can be used to set the various options.
+    wxTextProofOptions& SpellCheck(bool enable = true)
+    {
+        m_EnableSpellCheck = enable;
+        return *this;
+    }
+
+    wxTextProofOptions& GrammarCheck(bool enable = true)
+    {
+        m_EnableGrammarCheck = enable;
+        return *this;
+    }
+
+    wxTextProofOptions& Language(const wxString& lang)
+    {
+        m_lang = lang;
+        return *this;
+    }
+
+    // And the corresponding accessors.
+    bool IsSpellCheckingEnabled() const { return m_EnableSpellCheck; }
```

Comment:
Please can this be `IsSpellCheckEnabled()` to be consistent with the enabling function (`SpellCheck()`)

Reply:
OK, I'll rename all of them as suggested.